source/create.ts

```diff
@@ -120,13 +120,14 @@ const create = (defaults: InstanceDefaults): Got => {
 			) as GotReturn;
 		};
 
-		// TODO: remove this in Got 12
+		// TODO: Remove this in Got 12.
 		if (is.plainObject(url)) {
 			const mergedOptions = {
 				...url as Options,
 				...options
 			};
 
+			// @ts-expect-error
```

Comment:
Why is this needed?

Reply:
It didn't bother wasting time on it as that code is going away in the next major anyway.

Error:

```
source/create.ts:130:32 - error TS2322: Type '(string & Record<string, unknown>) | (URL & Record<string, unknown>)' is not assignable to type 'Options | Defaults | undefined'.
  Type 'string & Record<string, unknown>' is not assignable to type 'Options | Defaults | undefined'.
    Type 'string & Record<string, unknown>' is not assignable to type 'Options'.
      Types of property 'search' are incompatible.
        Type '{ (regexp: string | RegExp): number; (searcher: { [Symbol.search](string: string): number; }): number; }' is not assignable to type 'string'.

130  		setNonEnumerableProperties([url, options], mergedOptions);
```